test/common.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 Searis AS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/clarify/clarify-go"
+	"github.com/clarify/clarify-go/fields"
+)
+
+const (
+	AnnotationKey   = "there-is-always-money"
+	AnnotationValue = "in-the-banana-stand"
+)
+
+func getDefaultTimeRange() (time.Time, time.Time) {
+	tidenesMorgen := time.Unix(0, 0).UTC().Truncate(time.Hour)
+	tidenesKveld := tidenesMorgen.Add(10 * time.Hour)
+
+	return tidenesMorgen, tidenesKveld
+}
+
+func testName() string {
+	pc := make([]uintptr, 10)
+	n := runtime.Callers(3, pc)
+	frame, _ := runtime.CallersFrames(pc[:n]).Next()
+	funks := strings.Split(frame.Function, ".")
+	funk := funks[len(funks)-1]
+
+	return funk
+}
+
+func createPrefix() string {
+	test := testName()
+
+	return test + "/"
+}
+
+func getCredentials(t *testing.T) *clarify.Credentials {
+	var creds *clarify.Credentials
+	var err error
+
+	username := os.Getenv("CLARIFY_USERNAME")
+	password := os.Getenv("CLARIFY_PASSWORD")
+	endpoint := os.Getenv("CLARIFY_ENDPOINT")
+
+	creds = clarify.BasicAuthCredentials(username, password)
+
+	if endpoint != "" {
+		creds.APIURL = endpoint
+	}
+
+	if creds.Validate() != nil {
+		creds, err = clarify.CredentialsFromFile("clarify-credentials.json")
+		if err != nil {
+			t.Skip("no credentials found, skipping integration tests")
+		}
+	}
```

Comment:
Pretty sure I have commented in this previously; we should read the credentials file name from the environment; so that we don't "accidentally" run the tests (e.g. on localhost) based on you having a file with name `"clarify-credetials.json"` in your pwd.